src/sidebar/components/sidebar-content-error.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const { Fragment, createElement } = require('preact');
+const propTypes = require('prop-types');
+
+/**
+ * An error message to display in the sidebar.
+ */
+function SidebarContentError({ errorMessage, errorDetails, onLogin, auth }) {
+  return (
+    <Fragment>
+      <div className="annotation-unavailable-message">
+        <div className="annotation-unavailable-message__icon" />
+        <p className="annotation-unavailable-message__label">
+          {auth.status === 'logged-out' ? (
+            <span>
+              {errorMessage}
+              <br />
+              <a> You may need to </a>
+              <a className="loggedout-message__link" href="" onClick={onLogin}>
+                log in{' '}
+              </a>
+              <a>to see it.</a>
+            </span>
+          ) : (
+            <span>{errorDetails}</span>
+          )}
+        </p>
+      </div>
+    </Fragment>
+  );
+}
+
+SidebarContentError.propTypes = {
+  /* A short error message to be displayed when logged out along with a login prompt. */
+  errorMessage: propTypes.string,
+  /* A detailed error message explaining why the error message happened when logged in. */
+  errorDetails: propTypes.string,
```

Comment:
Hmm, the names here imply that the "details" message is longer and "errorMessage" is just a short summary, but that isn't really the case. The difference is just which one gets shown depending on whether the user is logged-in or not. Perhaps `loggedOutMessage` and `loggedInMessage` might be more obvious?

Reply:
I debated this for a long time. I wasn't sure what made sense. I was concerned loggedOutMessage vs loggedInMessage wasn't entirely accurate because technically loggedOutMessage is only the first part of the message. I thought it might be confusing to set loggedOutMessage and then see some additional text there that you didn't set. It's also a little odd to me that the short error message doesn't appear when the user is logged in, just the detailed description but I'm not sure I wanna open up that can of worms. If those names make sense to you though I'm happy to use them.